### PR TITLE
Update READMEs for Onboarding

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Update Envoyinit README and hack script

# Context

Learnings from onboarding conversations 

### Envoyinit Testing
Included testing notes in the envoyinit readme

### Basic Envoy Config
We decided that documented the yaml in the example config makes it easier to understand what different aspects are doing

### Update Envoy Version in Envoyinit Readme
Ensure that the readme references a more recently published version than 1.11.11

### Makefile
I introduced some GINKGO_ENV vars which exist in s-p and mask noisy errors/warnings. I also introduced a target with the desired ginkgo flags (randomize).

### CI Readme
Some common questions/gaps about the automated jobs in our CI pipeline are called out here

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by sam-heilbron** [comment]
Upgrade tests are not required and are known to be failing

**Comment by sam-heilbron** [comment]
I'm going to leave the `work in progress` label on to avoid the PR merging, until all 3 reviewers have had a chance to look at the changes, test them out and approve them, since it's additions to our onboarding material

File: projects/envoyinit/README.md [hunk 1]
```diff
@@ -52,21 +45,19 @@ In Gloo Edge, Envoy configuration is processed in the following order:
 4. The transformed configuration is provided to the Envoy executable
 
 ### Dynamic
-
 Envoy receives dynamic configuration via the [xDS protocol](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#xds-protocol). The Gloo [xDS](https://github.com/solo-io/gloo/tree/master/projects/gloo/pkg/xds) package contains relevant code for serving dynamic configuration.
 
 ## Debug
-
 It can be useful to run the Envoy proxy, without the control-plane, as a way of validating proxy behavior. [hack/envoy.yaml](./hack/envoy.yaml) provides example bootstrap that can be used. If you built an image you want to use, be sure to update container parameter in the command with the new image address output in the build step.
```
Comment (sheidkamp) [inline projects/envoyinit/README.md:51]: Maybe a note here that while the make/build step is run from the root directory, the `docker run` command is run from the project directory

Comment (sam-heilbron) [reply projects/envoyinit/README.md:51]: Yeah great call, I'll update

**Review by sheidkamp — COMMENTED** [review]
One comment for another README clarification. Example still works!

**Review by jmassasolo — APPROVED** [review]

**Review by sheidkamp — APPROVED** [review]

**Review by irateswami — APPROVED** [review]

**Pull request merged**